policy-controller/k8s/index/src/outbound/index/unmeshed_network.rs

```diff
@@ -0,0 +1,230 @@
+
+// === impl MatchedUnmeshedNetwork ===
+
+impl Networks {
+    pub fn find_match(&self, ip: IpAddr) -> Option<Cidr> {
+        let ip: Cidr = ip.into();
+        self.0
+            .iter()
+            .filter(|c| c.contains(&ip))
+            .min_by(|a, b| a.size().cmp(&b.size()))
+            .cloned()
+    }
+}
+
+// === impl UnmeshedNetwork ===
+
+impl From<linkerd_k8s_api::UnmeshedNetwork> for UnmeshedNetwork {
+    fn from(u: linkerd_k8s_api::UnmeshedNetwork) -> Self {
+        let name = u.name_unchecked();
+        let namespace = u
+            .namespace()
+            .expect("UnmeshedNetwork must have a namespace");
+
+        UnmeshedNetwork {
+            name,
+            namespace,
+            networks: Networks(u.spec.networks.clone()),
+            creation_timestamp: u.creation_timestamp().map(|d| d.0),
+        }
+    }
+}
+
+// === impl MatchedUnmeshedNetwork ===
+
+impl From<(Cidr, &UnmeshedNetwork)> for MatchedUnmeshedNetwork {
+    fn from((cidr, unet): (Cidr, &UnmeshedNetwork)) -> Self {
+        Self {
+            name: unet.name.clone(),
+            namespace: unet.namespace.clone(),
+            matched_cidr: MatchedCidr(cidr),
+            creation_timestamp: unet.creation_timestamp,
+        }
+    }
+}
+
+impl std::cmp::PartialOrd for MatchedUnmeshedNetwork {
+    fn partial_cmp(&self, other: &Self) -> Option<std::cmp::Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+impl std::cmp::Ord for MatchedUnmeshedNetwork {
+    fn cmp(&self, other: &Self) -> std::cmp::Ordering {
+        self.matched_cidr
+            .cmp(&other.matched_cidr)
+            .then_with(|| self.creation_timestamp.cmp(&other.creation_timestamp))
+            .then_with(|| self.namespace.cmp(&other.namespace).reverse())
+            .then_with(|| self.name.cmp(&other.name).reverse())
+    }
+}
+
+// === impl MatchedCidr ===
+
+impl std::cmp::PartialOrd for MatchedCidr {
+    fn partial_cmp(&self, other: &Self) -> Option<std::cmp::Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+impl std::cmp::Ord for MatchedCidr {
+    fn cmp(&self, other: &Self) -> std::cmp::Ordering {
+        let cidr_size_self = self.0.size();
+        let cidr_size_other = other.0.size();
+
+        match cidr_size_self.cmp(&cidr_size_other) {
+            std::cmp::Ordering::Less => std::cmp::Ordering::Greater,
+            std::cmp::Ordering::Greater => std::cmp::Ordering::Less,
+            std::cmp::Ordering::Equal => std::cmp::Ordering::Equal,
+        }
+    }
+}
+
+// Attempts to find the best matching network for a certain discovery look-up.
+// Logic is:
+// 1. if there are Unmeshed networks in the source_namespace, only these are considered
+// 2. the target IP is matches against the cidrs of the UnmeshedNetwork
+// 3. ambiguity is resolved as:
+//    - prefer the more specific cidr match
+//    - prefer older resource
+//    - if all fails, rely on alphabetical sort of namespace/name
+pub(crate) fn resolve_unmeshed_network<'n>(
```

Comment:
this could/should be an `impl Index` block in this file, instead of defining a `pub(crate)` function we redirect to above in `policy-controller/k8s/index/src/outbound/index.rs`.

Reply:
Sounds like a good suggestion. I want to avoid having to construct a full Index object just to test this function here. What would be a good way to do that? tests for context: https://github.com/linkerd/linkerd2/pull/13150/files#diff-959b6fe891f84b343517e0962d18ce42568f8a4214c87e6fc04eebe92d328ec1R134